### ISSUE TITLE
[TW-1780] Importing malformed annotation (without entry timestamp) causes segmentation fault

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1804]
_David Badura on 2016-03-12T09:40:57Z says:_ 

 When i import a task with annotations i get a segmentation fault. But only if the entry key in an annotation is missing.

Success:

```
 $ task import --
Importing 'STDIN'
{"description":"Buy the milk","annotations":[{"entry":"20160312T093133Z","description":"and Cheese"}]}

 add  2dc3282d-0a31-4229-bad8-fb1adb917b9c Buy the milk
Imported 1 tasks.
```

Segmentation Fault:

```
$ task import --
Importing 'STDIN'
{"description":"Buy the milk","annotations":[{"description":"and Cheese"}]}

[1]    6910 segmentation fault (core dumped)  /var/www/Taskwarrior/task import --
```

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2016-03-12T09:40:57Z
Modified: 2017-01-16T21:21:57Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2016-03-12T10:35:58Z says:_ 

 This should not segfault.

Then again, importing an incorrectly structured task should fail.

**Comment by taskwarrior** [comment]
_David Badura on 2016-03-12T10:42:10Z says:_ 

 Why is this incorrect? I imported this and i got a segmentation fault:

```
{"description":"Buy the milk","annotations":[{"description":"and Cheese"}]}
```

i have found a related issue: https://bug.tasktools.org/browse/TW-547

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2016-03-12T15:39:33Z says:_ 

 You said it yourself - "entry" is missing.

**Comment by taskwarrior** [comment]
_David Badura on 2016-03-12T17:06:33Z says:_ 

 With "missing" i meant, i didn't add this information because i don't necissarily have it when i create a new task with an annotation. why do i have to add this? shouldn't  taskwarrior create the timestamp automatically like "entry" property in Task entity?

For example:

```
1.  import new task
task import -- 
{"description":"Buy the milk"}
add  7f75a753-eed3-4d18-9a21-16edb230377d Buy the milk
1.  export task, taskwarrior created entry date automatically
task export 7f75a753-eed3-4d18-9a21-16edb230377d
[
{"id":56,"description":"Buy the milk","entry":"20160312T170008Z","modified":"20160312T170008Z","status":"pending","uuid":"7f75a753-eed3-4d18-9a21-16edb230377d","urgency":0}
]
1.  import new task with annotation
task import -- 
{"description":"Buy the milk","annotations":[{"description":"and Cheese"}]}
1.  segmentation fault, because the entry key in annotation is missing
```

when i use the "annotate" command, i didn't have to add an entry timestamp:

```
$ task add Buy the milk
$ task 1 annotate and Bread
$ task 1 annotate and Cheese
```

shouldn't it behave the same?

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2016-03-13T12:20:57Z says:_ 

 ```
i didn't add this information because i don't necissarily have it
```
I'm not sure there is ever a case when you don't know the time.  Add the `entry` value yourself.

```
why do i have to add this?
```
You found a segfault bug, and this is how you work around it.

```
shouldn't it behave the same?
```
Not according to the spec: http://taskwarrior.org/docs/design/task.html#attr_annotation

This is a bug report about a segfault caused by one of the JSON::*::dump methods.  When composing the error that would have told you that `entry` is missing, it crashes because it cannot serialize broken JSON.  It's a good bug.

If you additionally believe the JSON format should be relaxed to accommodate a missing annotation entry time, then submit a compelling feature request and/or complete patches.

**Comment by tbabej** [comment]
Reproduced on 2.6.0.

**Comment by tbabej** [comment]
I think relaxing the requirements here would be relatively easy and improve the user experience.

**Comment by tbabej** [comment]
@DavidBadura this is now fixed on 2.6.0. Thanks for reporting!
